core/unit_test/cuda/TestCuda_InterOp_StreamsMultiGPU.cpp

```diff
@@ -0,0 +1,49 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <TestCuda_Category.hpp>
+#include <Test_InterOp_Streams.hpp>
+
+namespace {
+TEST(cuda, multi_gpu) {
+  Kokkos::initialize();
+
+  int n_devices;
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaGetDeviceCount(&n_devices));
+
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaSetDevice(0));
+  cudaStream_t stream0;
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaStreamCreate(&stream0));
+
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaSetDevice(n_devices - 1));
+  cudaStream_t stream;
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaStreamCreate(&stream));
```

Comment:
Did you mean to mess with the CUDA runtime after `Kokkos::initialize()`?

Reply:
We are trying to make sure that we are always using the correct device even if the user changed it between `Kokkos` API calls.